consensus/misc/eip1559.go

```diff
@@ -31,79 +29,33 @@ import (
 // - basefee check
 func VerifyEip1559Header(config *params.ChainConfig, parent, header *types.Header) error {
```

Comment:
I tried to remove this function (moving the check outside) and found the logic would be simpler in removing places (and of course, we can also keep it like this).